alteryx/promote-r-client
# US30717: Add promote.sh to build/deploy

This PR adds the ability to let an advanced user supply a custom bash script with their model. The only restriction as of now is the script _must_ be named `promote.sh`. The user can assign the path of the script to a variable named `promotesh` in their model if they want to use the script. The script runs in the container setup and gets copied to `src/`.

**Dependencies:**
- https://github.com/alteryx/promote-python/pull/15
- https://den-dev-git-01.extendthereach.com/promote/promote-app/merge_requests/5

**To Test:**
- Switch to branch `promotesh` in promote-app and promote-python
- Add [this promote.sh script](https://gist.github.com/ErikAbrahamson/7b531d8b7ca796e492abd2cb363142a5) to the base folder of an example model
- Deploy a model and watch the build output stream. You should see the output from the promote.sh file during the build
- Ensure the model gets built successfully, and errors out if there are any errors (there shouldn't be)
- Deploy a separate model with no promote.sh script. The model should deploy as usual
